server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java

```diff
@@ -383,8 +377,27 @@ private CommandExecutor startCommandExecutor(
         return executor;
     }
 
+    @Nullable
+    private SessionManager initializeSessionManager() {
+        final AuthConfig authCfg = cfg.authConfig();
+        if (authCfg == null) {
+            return null;
+        }
+        SessionManager manager;
+        try {
+            manager = new FileBasedSessionManager(new File(cfg.dataDir(), "_sessions").toPath(),
+                                                  authCfg.sessionValidationSchedule());
+            manager = new CachedSessionManager(manager, Caffeine.from(authCfg.sessionCacheSpec()).build());
+            return new ExpiredSessionDeletingSessionManager(manager);
```

Comment:
This can leave the `SessionManager` instances involved here into an unclean state if the instantiation of any `SessionManager`s fails. How about storing the three session managers in separate local variables and cleaning them up if there was any failure?

Reply:
Fixed. But I just closed the `manager` instance when any exception is caught, because only FileBasedSessionManager has `close()` implementation and the other managers just forward `close()` call to their delegate. Does it make sense?

Reply:
Yes, but please use this pattern instead:

```java
boolean success = false;
try {
    ...
    manager = new ExpiredSessionDeletingSessionManager(manager);
    success = true;
    return manager;
} finally {
    if (!success) { ... close ... }
}
```